crates/ruma-api/src/lib.rs

```diff
@@ -591,3 +595,82 @@ impl MatrixVersion {
         }
     }
 }
+
+/// A specifier for what path variant of an endpoint to request.
+#[derive(Clone, PartialEq, Eq, Debug)]
+#[allow(clippy::exhaustive_enums)]
```

Comment:
This should be non-exhaustive, unconditionally even. There is no reason for outside code to match on this at all.